aips/ristretto_bulletproofs_elgamal.md

```diff
@@ -0,0 +1,75 @@
+
+## Impact
+
+The above benefits aside, this will marginally increase the Move standard library size. 
+
+## Rationale
+
+Bulletproofs is a simple and efficient protocol for range proofs over private values. It has also been well-studied and adapted in the literature, making it likely to be secure. 
+
+ElGamal encryption done in the exponent allows for homomorphic addition of ciphertexts, i.e. encrypted values can be added together without revealing them. 
+
+Ristretto255 combines the efficiency of non-prime order curves with the ease of implementation at the protocol level of prime order curves. This makes it ideal for many applications. Adding functions for point cloning and double scalar multiplications makes the pre-existing module easier to use. 
+
+
+## Specification
+
+Both the new modules and the additions to the Ristretto255 module have already been implemented:
+
+[Ristretto255](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255.move)
+
+[Bulletproofs](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255_bulletproofs.move)
+
+[ElGamal Encryption](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255_elgamal.move)
+
+[Pedersen Commitments](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255_pedersen.move)
+
+The Ristretto255 module is implemented using the dalek cryptography [curve25519 crate](https://github.com/dalek-cryptography/curve25519-dalek).
+
+The Bulletproofs module is implemented using the dalek cryptography [bulletproofs crate](https://github.com/dalek-cryptography/bulletproofs).
+
+The ElGamal encryption and Pedersen commitment modules are implemented without Move natives over the Ristretto255 Move module referenced above. 
+
+## Risks and Drawbacks
+
+Future obsolescence is often a risk when implementing cryptography primitives. As of writing this, both Ristretto255 and Bulletproofs have been in use for several years, making their continued used in the immediate future more likely than not. ElGamal encryption and Pedersen commitments have been in use for multiple decades.
+## Timeline
+
+### Suggested developer platform support timeline
+
+Describe the plan to have SDK, API, CLI, Indexer support for this feature is applicable. 
```

Comment:
Is this applicable? I suspect not really.

Reply:
It is not. Will clarify.